### PR TITLE
Deploy `PurchaseTester` for `macOS`

### PR DESCRIPTION
**Pull request by NachoSoto**
Follow up to #2003. Also submitting `macOS` build for [CSDK-510].

Changed this from `Catalyst` to native `macOS` because of this issue: https://github.com/fastlane/fastlane/issues/20826

App_store_512

[CSDK-510]: https://revenuecats.atlassian.net/browse/CSDK-510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by NachoSoto** [comment]
This is pretty much ready, but it's blocked by https://github.com/fastlane/fastlane/issues/20826.

**Comment by NachoSoto** [comment]
I'm thinking of resuscitating this and submitting a native `macOS` build instead (thanks to #2175).

File: Tests/TestingApps/PurchaseTesterSwiftUI/PurchaseTester.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -675,30 +681,36 @@
...
 				INFOPLIST_KEY_UIApplicationSupportsIndirectInputEvents = YES;
 				INFOPLIST_KEY_UILaunchScreen_Generation = YES;
 				INFOPLIST_KEY_UISupportedInterfaceOrientations_iPad = "UIInterfaceOrientationPortrait UIInterfaceOrientationPortraitUpsideDown UIInterfaceOrientationLandscapeLeft UIInterfaceOrientationLandscapeRight";
 				INFOPLIST_KEY_UISupportedInterfaceOrientations_iPhone = "UIInterfaceOrientationPortrait UIInterfaceOrientationLandscapeLeft UIInterfaceOrientationLandscapeRight";
-				LD_RUNPATH_SEARCH_PATHS = (
-					"$(inherited)",
-					"@executable_path/Frameworks",
-				);
+				LD_RUNPATH_SEARCH_PATHS = "@executable_path/Frameworks";
+				"LD_RUNPATH_SEARCH_PATHS[sdk=macosx*]" = "@executable_path/../Frameworks";
```
Comment (NachoSoto) [inline Tests/TestingApps/PurchaseTesterSwiftUI/PurchaseTester.xcodeproj/project.pbxproj:699]: These crashes:
> Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: @rpath/RevenueCat_76EF69325FE509FC_PackageProduct.framework/Versions/A/RevenueCat_76EF69325FE509FC_PackageProduct
Referenced from: <DEACE43D-F7AB-307B-A35E-44EC03198E29> /Applications/PurchaseTester.app/Contents/MacOS/PurchaseTester
Reason: tried: '/Applications/PurchaseTester.app/Contents/MacOS/Frameworks/RevenueCat_76EF69325FE509FC_PackageProduct.framework/Versions/A/RevenueCat_76EF69325FE509FC_PackageProduct' (no such file), '/Applications/PurchaseTester.app/Contents/MacOS/Frameworks/RevenueCat_76EF69325FE509FC_PackageProduct.framework/Versions/A/RevenueCat_76EF69325FE509FC_PackageProduct' (no such file), '/System/Volumes/Preboot/Cryptexes/OS@rpath/RevenueCat_76EF69325FE509FC_PackageProduct.framework/Versions/A/RevenueCat_76EF69325FE509FC_PackageProduct' (no such file), '/Applications/PurchaseTester.app/Contents/MacOS/Frameworks/RevenueCat_76EF69325FE509FC_PackageProduct.framework/Versions/A/RevenueCat_76EF69325FE509FC_PackageProduct' (no such file), '/Applications/PurchaseTester.a

Were due to this being wrong. `macOS` has a slightly different folder structure.
@aboedo this was the solution.

**Comment by NachoSoto** [comment]
This is ready 🎉

File: fastlane/Fastfile [hunk 2]
```diff
@@ -510,6 +512,28 @@ platform :ios do
       submit_for_review: false,
       run_precheck_before_submit: false
     )
+
+    # macOS
```
Comment (tonidero) [inline fastlane/Fastfile:516]: Should we separate into 2 functions?

**Review by tonidero — APPROVED** [review]
LGTM!

File: fastlane/Fastfile (hunk 2 above)
Comment (NachoSoto) [reply fastlane/Fastfile:516]: Sounds good 👍🏻

**Pull request merged**